torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -327,6 +327,13 @@ void compileKernel(CudaKernel* entry) {
   std::tie(func_name, code) = codeGeneration(entry->fusion_.get());
 
   static int32_t compiled_kernel_id = 0;
+  const char* debug_env = getenv("PYTORCH_CUDA_FUSER_DEBUG");
+  if (debug_env && atoi(debug_env)) {
+    std::cout << "==== codegen output for kernel: " << compiled_kernel_id++
```

Comment:
1. I'm a bit nervous about `compiled_kernel_id++` part: incrementing it part of, and only if, a debug print seems suspicious. What is the intention here?

2. nit: for printing debug dumps I find it useful to start with a new line (ex. std::cout << std::endl or just "\n====...". This helps when the previous output is an incomplete line.

Reply:
1. Good catch! This is actually a bug.
The `compiled_kernel_id` is literally just for debugging purposes, it is used in order to map printed out kernel to the binary/ptx file we spit out (if we choose to). But I need to move it the increment outside of the condition, as we could have strange combination of env flags.

2. Will do.